core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -212,14 +212,10 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
                 URI uri = (URI) pathJDK7.getMethod("toUri").invoke(toPath);
 
                 baseResourceURL = uri.toURL();
-            } catch (NoSuchMethodException e) {
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
```

Comment:
Indeed, I had actually in mind to do another pass to look for pre-java7 specific cases and file a new PR.

Btw, there are also cases where potentially `ReflectiveOperationException` might be used instead of the typical code to catch all the reflection related exceptions, but might be harder to be sure to be backward-compatible.